### ISSUE TITLE
net5 tfm: "hydrating" empty TargetPlatformVersions

### ISSUE THREAD

**Issue by zkat** [NuGet/Home#9444]
NuGet itself doesn't have the information it needs in order to understand what version `net5.0-ios` and such might mean, but omitting versions is an important part of this feature.

After discussions with the dotnet and msbuild teams, my understanding is that they will "fill in" the `NuGetFramework` instances with the actual version using some API call that's yet to be determined (or, preferably, hand us a new one, so we can keep `NuGetFramework` immutable).

**Comment by dsplaisted** [comment]
We will be splitting the `NuGetFramework` into the following MSBuild properties:

- `TargetFrameworkIdentifier`
- `TargetFrameworkVersion`
- `TargetFrameworkProfile` (this is for completeness, we don't expect this to be used normally)
- `TargetPlatformIdentifier`
- `TargetPlatformVersion`

If the platform version isn't specified in the `NuGetFramework`, then the `TargetPlatformVersion` will initially be unset, and the targets for that platform will set it to a default value.

NuGet should be reading from the 5 component properties in various scenarios, and that's how NuGet should be picking up what the target platform version is.

Here's an issue tracking doing this in the targets for Windows: https://github.com/dotnet/sdk/issues/11233

**Comment by nkolev92** [comment]
Related to https://github.com/NuGet/Home/issues/9443 and probably should be considered under https://github.com/NuGet/Home/issues/5154

**Comment by nkolev92** [comment]
Reading the target frameworks in the inner build will make this possible.

**Comment by zkat** [comment]
This will be fairly straightforward once https://github.com/nuget/home/issues/9756 is implemented.

**Comment by nkolev92** [comment]
https://github.com/NuGet/Home/issues/9756 is merged, so this can probably be closed now. 

It might be nice to grab the latest SDK and verify the scenario once the NuGet insertion that should be started tomorrow completes.

**Comment by zkat** [comment]
This is likely already done, and just needs to be tested against dotnet's changes.

tl;dr, we need to check that when we multi-target net5.0 and net5.0-windows, that we have two different frameworks and we don't have a random "windows-7.0".

Closing this in favor of https://github.com/NuGet/Home/issues/9873
